modules/core/src/main/java/org/apache/ignite/internal/managers/indexing/IndexesRebuildTask.java

```diff
@@ -84,17 +87,20 @@
         // To avoid possible data race.
         GridFutureAdapter<Void> outRebuildCacheIdxFut = new GridFutureAdapter<>();
 
-        // An internal future for the ability to cancel index rebuilding.
-        // This behavior should be discussed in IGNITE-14321.
         IgniteLogger log = cctx.kernalContext().grid().log();
 
+        // An internal future for the ability to cancel index rebuilding.
         SchemaIndexCacheFuture intRebFut = new SchemaIndexCacheFuture(new SchemaIndexOperationCancellationToken());
-        cancelIndexRebuildFuture(idxRebuildFuts.put(cctx.cacheId(), intRebFut), log);
+
+        SchemaIndexCacheFuture prevIntRebFut = idxRebuildFuts.put(cctx.cacheId(), intRebFut);
```

Comment:
have a call, seems we can simplify here, i huge +1 for such simplification here.

Reply:
I suggest leaving it as it is, since **GridQueryProcessor#markAsRebuildNeeded** should only be called on rebuilds of all indexes and **SchemaIndexCacheVisitorImpl** can be used to add a new index in https://github.com/apache/ignite/blob/020ce49a34a92d9fe98b668725df534c3fbd404f/modules/core/src/main/java/org/apache/ignite/internal/processors/query/GridQueryProcessor.java#L1895.